widgetssdk/src/main/java/com/glia/widgets/core/locales/CachedFileNamesUseCase.kt

```diff
@@ -0,0 +1,31 @@
+package com.glia.widgets.core.locales
+
+import android.content.SharedPreferences
+import com.glia.widgets.helper.Logger
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import io.reactivex.Completable
+import io.reactivex.Single
+
+class CachedFileNamesUseCase(private val sharedPrefs: SharedPreferences, private val gson: Gson) {
+
+    fun store(locale: String): Completable {
+        return retrieve().map { map ->
+            map.remove(locale)
+            map[locale] = System.currentTimeMillis()
```

Comment:
The SharedPreferences is a Key-Value storage, and I think in this case it would be better to store locale-time directly in Preferences instead of wrapping it in Map than serializing/deserializing to JSON.
So in this case, the key could be `"com.glia.widgets.core.locales.$locale"`